app/javascript/mastodon/features/compose/util/counter.js

```diff
@@ -0,0 +1,7 @@
+const urlPlaceholder = 'xxxxxxxxxxxxxxxxxxxxxxx';
+
+export function countableText(inputText) {
+  return inputText
+    .replace(/https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/g, urlPlaceholder)
```

Comment:
The 256 part is not a big issue, as all domain names are [limited to 255 characters in practice](https://stackoverflow.com/a/28918017/539465).
However, the regexp for the TLD is `[a-z]{2,6}`, which wrong because TLDs are not limited to 6 characters. eg. `.network` (used by mastodon.network)